### PR TITLE
added parameters

### PR DESCRIPTION
**Pull request by ankur-dwivedi**
https://www.loom.com/share/493418d547e8481191895cad2f238be2?sid=d5842773-c2cc-40b2-b05c-c789b065a9e6

### CODE REVIEW

**Review by ahnv — APPROVED** [review]

File: Imagekit/Models/UploadTransformation.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+namespace Imagekit.Models
+{
+    using System.Collections.Generic;
+
+    public class PostTransformation
+    {
+        public string type { get; set; }
+        public string value { get; set; }
+    }
```
Comment (ahnv) [inline Imagekit/Models/UploadTransformation.cs:9]: Documentation for pre and post transform missing in README.md

File: Imagekit/Helper/TransformationTypes.cs [hunk 2]
```diff
@@ -407,6 +218,18 @@ public Transformation EffectGray()
...
+    {
+        return this.Add("e-shadow", "true");
+    }
+
+    /// <summary></summary>
+    public Transformation EffectGradient()
+    {
+        return this.Add("e-gradient", "true");
+    }
+
```
Comment (ahnv) [inline Imagekit/Helper/TransformationTypes.cs:232]: Please document these transformations also in Readme.md also

**Review by ahnv — CHANGES_REQUESTED** [review]

File: Imagekit/Models/UploadTransformation.cs (hunk 1 above)
Comment (ankur-dwivedi) [reply Imagekit/Models/UploadTransformation.cs:9]: done

File: Imagekit/Helper/TransformationTypes.cs (hunk 2 above)
Comment (ankur-dwivedi) [reply Imagekit/Helper/TransformationTypes.cs:232]: done

File: Imagekit/Helper/TransformationTypes.cs [hunk 3]
```diff
@@ -407,6 +218,18 @@ public Transformation EffectGray()
         return this.Add("e-grayscale", "true");
     }
 
+    /// <summary></summary>
+    public Transformation EffectShadow()
+    {
+        return this.Add("e-shadow", "true");
```
Comment (ahnv) [inline Imagekit/Helper/TransformationTypes.cs:224]: There is no such transformation as e-shadow-true, e-shadow has a syntax like this e-shadow-<blur>_<saturation>_<x-offset>_<y-offset> or just e-shadow 
https://imagekit.io/docs/effects-and-enhancements#shadow---e-shadow

File: Imagekit/Helper/TransformationTypes.cs [hunk 4]
```diff
@@ -407,6 +218,18 @@ public Transformation EffectGray()
...
+    }
+
+    /// <summary></summary>
+    public Transformation EffectGradient()
+    {
+        return this.Add("e-gradient", "true");
```
Comment (ahnv) [inline Imagekit/Helper/TransformationTypes.cs:230]: There is no such transformation as e-gradient-true, e-shadow has a syntax like this `e-gradient-<linear-direction>_<from-color>_<to-color>_<stop-point>` or just e-gradient
https://imagekit.io/docs/effects-and-enhancements#gradient---e-gradient

File: README.md [hunk 5]
```diff
@@ -91,7 +93,7 @@ string imageURL = imagekit.Url(trans).Path(path).TransformationPosition("query")
 This results in a URL like
 
 ```plaintext
-https://ik.imagekit.io/default_image.jpg?tr=w-400%2Ch-300%2Car-4-3%2Cq_40%2Cc-force%2Ccm-extract%2Cfo-left%2Cf-jpeg%2Ch-200%2Cw-300%2Cl-image%2Ci-logo.png%2Cl-end
+https://ik.imagekit.io/default_image.jpg?tr=w-400%2Ch-300%2Car-4-3%2Cq-40%2Cc-force%2Ccm-extract%2Cfo-left%2Cf-jpeg%2Ce-shadow-true%2Ce-gradient-true%2Ch-200%2Cw-300%2Cl-image%2Ci-logo.png%2Cl-end
```
Comment (ahnv) [inline README.md:96]: This url is invalid, please fix it

Comment (ankur-dwivedi) [reply README.md]: done

File: Imagekit/Helper/TransformationTypes.cs (hunk 4 above)
Comment (ankur-dwivedi) [reply Imagekit/Helper/TransformationTypes.cs]: Fixed it

File: Imagekit/Helper/TransformationTypes.cs (hunk 3 above)
Comment (ankur-dwivedi) [reply Imagekit/Helper/TransformationTypes.cs]: fixed it

**Pull request merged**